### ISSUE TITLE
[Select] Space to underlying form element is reduces when select option is selected 

### ISSUE THREAD

**Issue by FabianKoestring** [material-components/material-components-web#4895]
## Bug report
When a select option is selected the spacing to the underlying element is automatically changed. This bug is visible on the demo site on the outlined enhanced select element.
https://material-components.github.io/material-components-web-catalog/#/component/select

### Steps to reproduce
1. Go to https://material-components.github.io/material-components-web-catalog/#/component/select
2. Click on outlined enhanced select element
3. Choose option
4. Spacing to the underlying elements gets reduced

### Actual behavior
Spacing to the underlying element gets reduced.

### Expected behavior
The spacing should not be changed.

**Comment by CyborgSemon** [comment]
This is happening to me as well
Gif of issue: (look closely)
![selectBug](https://user-images.githubusercontent.com/26075792/61354099-7bc45980-a8c5-11e9-9bbc-0136f4cf0bf5.gif)

Images of inspect view of parent div:
<img width="251" alt="Screen Shot 2019-07-17 at 7 02 09 PM" src="https://user-images.githubusercontent.com/26075792/61354122-8bdc3900-a8c5-11e9-91d1-b9f54f99e29a.png">

You can see the 4px under the select when the value is empty

<img width="247" alt="Screen Shot 2019-07-17 at 7 02 21 PM" src="https://user-images.githubusercontent.com/26075792/61354125-90a0ed00-a8c5-11e9-9dc7-8dad5e841085.png">
Now that 4px is gone

The div I'm using is set up like this
```html
<div>
  <div>Some text</div>
  <div class="mdc-select">...</div>
</div>
```

**Comment by Darhakexge** [comment]
This happens because `.mdc-select__selected-text` not have child nodes and browsers add those extra 4px (I not have idea why).
To solve it, `.mdc-select__selected-text` needs at least one child node (it can be a pseudo element)

This rules solve the problem, also I made a [demo](https://codepen.io/Darhakexge/pen/VoWLWJ)
```css
.mdc-select__selected-text{
	display: flex; /*For filled select*/
}
.mdc-select__selected-text::before{
	content: "";
}
```

**Comment by FabianKoestring** [comment]
> This happens because `.mdc-select__selected-text` not have child nodes and browsers add those extra 4px (I not have idea why).
> To solve it, `.mdc-select__selected-text` needs at least one child node (it can be a pseudo element)
> 
> This rules solve the problem, also I made a [demo](https://codepen.io/Darhakexge/pen/VoWLWJ)
> 
> ```css
> .mdc-select__selected-text{
> 	display: flex; /*For filled select*/
> }
> .mdc-select__selected-text::before{
> 	content: "";
> }
> ```

Works perfectly. Will this be part of next release?

**Comment by Darhakexge** [comment]
I found a more optimal way to remove that extra space using the `vertical-align` property
```css
.mdc-select{
  vertical-align: middle;
}
```

**Comment by allan-chen** [comment]
This should be fixed after layout overhaul in 28d10a9, which includes changing `.mdc-select__anchor` from `display: inline-flex` to `display: flex` so that the root `mdc-select` element dictates width.
